v2/spiffetls/tlsconfig/config.go

```diff
@@ -73,14 +132,25 @@ func MTLSServerConfig(svid x509svid.Source, bundle x509bundle.Source, authorizer
 // wrapped by by this package and invoked after SPIFFE authentication has
 // completed.
 func HookMTLSServerConfig(config *tls.Config, svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) {
-	panic("not implemented")
+	config.InsecureSkipVerify = true
+	config.ClientAuth = tls.RequireAndVerifyClientCert
+	config.GetCertificate = GetCertificate(svid)
+	config.GetClientCertificate = GetClientCertificate(svid)
+	config.VerifyPeerCertificate = wrapVerifyPeerCertificate(config.VerifyPeerCertificate, VerifyPeerCertificate(bundle, authorizer))
+
+	// No used by server
```

Comment:
All of these are used by the server. `GetClientCertificate` can go here though once you address the above comment.